### PR TITLE
Add cliSetup command to test clusters configuration

### LINKED ISSUE #50382: Add ability for plugins to execute arbitrary commands before integ tests
**Describe the feature**: Before Elasticsearch 7.5 a plugin developer could execute arbitrary plugins before the integration test was run.

I use this functionality in one of my ingest plugins to execute one the commands in the `bin/` directory, that is doing some setup before running the integration test. This used to be done via the `setupCommands` task before 7.4

```
integTestCluster {
  setting 'ingest.opennlp.model.file.names',     'en-ner-persons.bin'
  setting 'ingest.opennlp.model.file.locations', 'en-ner-locations.bin'
  setting 'ingest.opennlp.model.file.dates',     'en-ner-dates.bin'

  setupCommand 'downloadModels', 'bin/ingest-opennlp/download-models'
}
```

With the cluster formation refactoring this ability seems to be gone in 7.5 and I do not see any obvious alternatives for plugin developers.

Is there any chance to add such functionality back or can we document a workaround?

### PR DESCRIPTION
**Pull request by rjernst**
This commit adds a cliSetup command that can be used to run arbitrary
bin scripts to setup a test cluster. This is the same as what was
previously called setupCommands in cluster formation tasks.

closes #50382

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java [hunk 1]
```diff
@@ -124,6 +124,7 @@
     private final LazyPropertyMap<String, CharSequence> settings = new LazyPropertyMap<>("Settings", this);
     private final LazyPropertyMap<String, CharSequence> keystoreSettings = new LazyPropertyMap<>("Keystore", this);
     private final LazyPropertyMap<String, File> keystoreFiles = new LazyPropertyMap<>("Keystore files", this, FileEntry::new);
+    private final LazyPropertyList<CliEntry> cliSetup = new LazyPropertyList<>("CLI setup commands", this);
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: We need to track this collection as an input here as well. Look below for examples of the `@Nested` entries.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java (hunk 1 above)
Comment (rjernst) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: I pushed a4a92c4

**Review by mark-vieira — APPROVED** [review]
LGTM 👍

**Comment by mark-vieira** [comment]
@rjernst I believe this is ready to merge.

**Pull request merged**